src/wp-includes/functions.php

```diff
@@ -7635,3 +7645,85 @@ function is_php_version_compatible( $required ) {
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @return string  Inline JavaScript code wrapped around <script> tags if $echo is true, null otherwise.
+ */
+function wp_sanitize_script_attributes( $attributes = array() ) {
+	if ( ! array_key_exists( 'type', $attributes ) && ! is_admin() && ! current_theme_supports( 'html5', 'script' ) ) {
+		$attributes['type'] = 'text/javascript';
+	}
+	/**
+	 * Filters script attributes.
+	 *
+	 * @param string $attributes    Associative array representing script attributes.
+	 */
+	$attributes = apply_filters( 'wp_script_attributes', $attributes );
+
+	$attributes_string = '';
+	// Only the attribute name is added to $attributes_string for entries with a boolean value, and that are true.
+	foreach ( $attributes as $attribute_name => $attribute_value ) {
+		if ( is_bool( $attribute_value ) ) {
+			if ( $attribute_value && '' !== $attribute_name ) {
+				$attributes_string .= ' ' . $attribute_name;
+			}
+		} else {
+			$attributes_string .= sprintf( ' %s="%s"', $attribute_name, esc_attr( $attribute_value ) );
+		}
+	}
+
+	return $attributes_string;
+}
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `wp_script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param array     $attributes     `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @param bool      $echo           Optional. Prints to the page if true, returns the value otherwise.
+ * @param string    $content        Optional. Code to be wrapped around `<script>` tags.
+ * @return string|null  Inline JavaScript code wrapped around `<script>` tags if $echo is true, null otherwise.
+ */
+function wp_print_script_loader_tag( $attributes, $echo = true ) {
+	$output = sprintf( "<script%s></script>\n", wp_sanitize_script_attributes( $attributes ) );
+
+	if ( $echo ) {
+		echo $output;
+	} else {
+		return $output;
+	}
+}
+
+/**
+ * Wraps inline JavaScript in `<script>` tags.
+ *
+ * Enables to inject attributes in the `<script>` tag via the `wp_script_attributes` filter.
+ * Automatically injects type attribute if needed.
+ *
+ * @param string    $javascript     Inline JavaScript code.
+ * @param array     $attributes     Optional. `<script>` tag attributes.
+ * @param bool      $double_quotes  Optional. Uses double quotes `"` to wrap attribute values if true, single quotes `'` otherwise.
+ * @param bool      $echo           Optional. Prints to the page if true, returns the value otherwise.
+ * @return string|null  Inline JavaScript code wrapped around `<script>` tags if $echo is true, null otherwise.
+ */
+function wp_print_inline_script_tag( $javascript, $attributes = array(), $echo = true ) {
```

Comment:
I would love to see a hook here so third party plugins, or even core, can generate the sha256 hash of the script, and pass it to the CSP header under `script-src`.

Reply:
Hi Ayesh!
Thank you for the suggestion, will try to implement it in the next few days.
This PR is specifically about the refactoring of inline event handlers and builds on top of #498, which is the PR about refactoring inline scripts.
Would you mind moving the conversation to the [Trac ticket](https://core.trac.wordpress.org/ticket/39941) related to this change?